perf/src/deduper.rs

```diff
@@ -248,7 +249,7 @@ mod tests {
         let mut packet = Packet::new([0u8; PACKET_DATA_SIZE], Meta::default());
         let mut dup_count = 0usize;
         for _ in 0..num_packets {
-            let size = rng.gen_range(0..PACKET_DATA_SIZE);
+            let size = random_u64_range(&mut rng, 0..PACKET_DATA_SIZE as u64) as usize;
```

Comment:
To confirm - this test would fail if we used rand 0.9.2 ?

We'd have to find the original PR (if there was one 😅 ) but I wouldn't be surprised if the values in the test case rows above were found empirically (ie `num_dups`). If that is indeed correct, there might be some validity to an argument that says "update the cases" or "update the tests so we don't rely on exactness from "random" output)

Reply:
Yes, in some of the first PRs I had to update it (https://github.com/anza-xyz/agave/pull/9036/commits/11d44fffbfc7a824176b69941e6f335f8647d30c), but you are right, the stability of those test cases doesn't seem to be important, since production code doesn't use breaking parts of `rand`, the way it uses randomness is stable:
```rust
fn new_random_state<R: Rng>(rng: &mut R) -> RandomState {
    RandomState::with_seeds(rng.gen(), rng.gen(), rng.gen(), rng.gen())
}
```

the final PR is already cooked and passing (https://github.com/anza-xyz/agave/pull/9276), so the question is if it will need to have those tests update or now. Could probably update them separately too by adding a pinned `rand=0.9.2` dependency there (dev-dependencies).
Let me know which course of action is preferable.